swig/cpp/examples/cpp_application_changes_example.cpp

```diff
@@ -147,16 +148,13 @@ module_change_cb(sr_session_ctx_t *session, const char *module_name, sr_notif_ev
         snprintf(change_path, MAX_LEN, "/%s:*", module_name);
 
         Subscribe subscribe(&sess);
-        subscribe.get_changes_iter(&change_path[0], &it);
-
-	while (true) {
-            try {
-                sr_change_oper_t oper = subscribe.get_change_next(&it, &old_value, &new_value);
-                print_change(oper, &old_value, &new_value);
-            } catch( const std::exception& e ) {
-                break;
-            }
+        it = subscribe.get_changes_iter(&change_path[0]);
+
+        while (oper = subscribe.get_change_next(it, old_value, new_value)) {
+		if (old_value->val() && new_value->val())
```

Comment:
This will work only for SR_OP_MODIFIED. If a change operation is created old value is NULL, similarly when if it is deleted new value is NULL.